File: tests/expectations/cfg_macro.cpp
#include <cstdarg>
#include <cstdint>
#include <cstdlib>
#include <ostream>
#include <new>

#if (defined(A) && defined(B_C))
#define __CBINDGEN_CFG_A2_D1_A_D3_B_C(...) __VA_ARGS__
#else
#define __CBINDGEN_CFG_A2_D1_A_D3_B_C(...)
#endif
#if (defined(A_B) && defined(C))
#define __CBINDGEN_CFG_A2_D3_A_B_D1_C(...) __VA_ARGS__
#else
#define __CBINDGEN_CFG_A2_D3_A_B_D1_C(...)
#endif

struct Collision {
#if (defined(A) && defined(B_C))
  int32_t x
#endif
  ;
#if (defined(A_B) && defined(C))
  int32_t y
#endif
  ;
};

struct Nested {
  Collision collision;
  const uint8_t *pointer;
};

struct Foo {
#if defined(PLATFORM_WIN)
  int32_t x
#endif
  ;
};

#define NESTED Nested{ /* .collision = */ Collision{ __CBINDGEN_CFG_A2_D1_A_D3_B_C(/* .x = */ 1,) __CBINDGEN_CFG_A2_D3_A_B_D1_C(/* .y = */ 2,) }, /* .pointer = */ (const uint8_t*)0 }

constexpr const Foo FOO = Foo{
#if defined(PLATFORM_WIN)
  /* .x = */ 0
#endif
};

File: tests/rust/cfg_macro.rs
#[repr(C)]
pub struct Collision {
    #[cfg(all(a, b_c))]
    pub x: i32,
    #[cfg(all(a_b, c))]
    pub y: i32,
}

#[repr(C)]
pub struct Nested {
    pub collision: Collision,
    pub pointer: *const u8,
}

pub const NESTED: Nested = Nested {
    collision: Collision {
        #[cfg(all(a, b_c))]
        x: 1,
        #[cfg(all(a_b, c))]
        y: 2,
    },
    pointer: 0 as *const u8,
};

#[repr(C)]
pub struct Foo {
    #[cfg(windows)]
    pub x: i32,
}

pub const FOO: Foo = Foo {
    #[cfg(windows)]
    x: 0,
};
